client/active-game/game-server.js

```diff
@@ -1,80 +1,97 @@
+        log.verbose(`No game connection for ${id}`)
+      }
+    })
 
-  async function onEnd(data, next) {
-    const body = data.get('body')
-    activeGameManager.handleGameEnd(data.get('gameId'), body.results, body.time)
+    server.on('connection', (socket, request) => {
+      const gameIdIndex = request.rawHeaders.indexOf('x-game-id')
+      if (gameIdIndex !== -1) {
+        const gameId = request.rawHeaders[gameIdIndex + 1]
+        log.verbose('game websocket connected')
+        socket.on('close', () => {
+          log.verbose('game websocket disconnected')
+          this.idToSocket.delete(gameId)
+        })
+        socket.on('message', message => {
+          this.onMessage(gameId, message)
+        })
+        this.idToSocket.set(gameId, socket)
+        activeGameManager.handleGameConnected(gameId)
+      }
+    })
   }
 
-  async function onReplaySave(data, next) {
-    const { path } = data.get('body')
-    activeGameManager.handleReplaySave(data.get('gameId'), path)
+  sendCommand(socket, command, payload) {
+    socket.send(
+      JSON.stringify({
+        command,
+        payload,
+      }),
+    )
   }
 
-  async function onWindowMove(data, next) {
-    const { x, y } = data.get('body')
-    dispatch(mergeLocalSettings({ gameWinX: x, gameWinY: y }))
+  onMessage(gameId, message) {
+    const { command, payload } = JSON.parse(message)
+    switch (command) {
+      case '/game/setupProgress':
+        activeGameManager.handleSetupProgress(gameId, payload.status)
+        break
+      case '/game/start':
+        activeGameManager.handleGameStart(gameId)
+        break
+      case '/game/end':
+        activeGameManager.handleGameEnd(gameId, payload.results, payload.time)
+        break
+      case '/game/replaySave':
+        activeGameManager.handleReplaySave(gameId, payload.path)
+        break
+      case '/game/windowMove':
+        const { x, y } = payload
+        dispatch(mergeLocalSettings({ gameWinX: x, gameWinY: y }))
+        break
+      default:
+        log.error(`Received an unknown command '${message.command}' from ${gameId}`)
+    }
   }
-
-  nydus.on('connection', socket => {
-    const id = socket.conn.request.headers['x-game-id']
-    subscribeToCommands(nydus, socket, id)
-    activeGameManager.handleGameConnected(id)
-  })
-
-  nydus.registerRoute('/game/setupProgress', getGameId, onSetupProgress)
-  nydus.registerRoute('/game/start', getGameId, onStart)
-  nydus.registerRoute('/game/end', getGameId, onEnd)
-  nydus.registerRoute('/game/replaySave', getGameId, onReplaySave)
-  nydus.registerRoute('/game/windowMove', onWindowMove)
-
-  activeGameManager.on('gameCommand', (id, command, payload) => {
-    log.verbose(`Sending game command to ${id}: ${command}`)
-    sendCommand(nydus, id, command, payload)
-  })
 }
 
 function makeGameServer() {
   if (!IS_ELECTRON) {
     return null
   }
 
-  const nydus = require('nydus').default
+  const WebSocket = require('ws/index.js')
```

Comment:
This import could really use a comment on why we need to import `index.js` directly (to workaround the `browser` field in `ws`'s `package.json` which webpack uses for Electron apps; *smh*).